src/main/java/com/metaphorce/security/service/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package com.metaphorce.security.service;
+
+
+import com.metaphorce.user.entities.UserEntity;
+import com.metaphorce.user.respositories.UserRepository;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+
+/**
+ * Service for loading user details from the database.
+ */
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+
+    private final UserRepository userRepository;
+
+    /**
+     * Constructs a UserDetailsServiceImpl with the given user repository.
+     *
+     * @param userRepository the user repository for accessing user data.
+     */
+    public UserDetailsServiceImpl(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * Loads the user details by username (email).
+     *
+     * @param email the email of the user.
+     * @return the UserDetails object containing user information.
+     * @throws UsernameNotFoundException if the user is not found.
+     */
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
```

Comment:
There is no necessity to set throws UsernameNotFoundException because this exception inherits from RuntimeException